### PR TITLE
Broadcast channel fixes

### PR DESCRIPTION
**Pull request by nidhi-bhanushali**
### Summary
Fixes for broadcast channel

### Task
https://app.asana.com/0/1206407507151849/1208156816232338/f

### CODE REVIEW

File: src/components/OAuthPopup.tsx [hunk 1]
```diff
@@ -47,16 +48,30 @@ export const OAuthPopup = ({
...
-					error: errorMessage,
-				});
+				console.error('message sent', errorMessage);
 			}
 		} else {
 			throw new Error('No BroadcastChannel support');
 		}
-	}, []);
+
+		const handleListener = (message: MessageEvent<TMessageData>) => {
```
Comment (Cypherball) [inline src/components/OAuthPopup.tsx]: define outside with useCallback

File: src/components/OAuthPopup.tsx [hunk 2]
```diff
@@ -21,9 +22,10 @@ export const OAuthPopup = ({
 		</div>
 	),
 }: Props) => {
-	const channel = new BroadcastChannel('oauth_channel');
+	const channel = useMemo(() => new BroadcastChannel('refrens_oauth_channel'), []);
```
Comment (Cypherball) [inline src/components/OAuthPopup.tsx]: useRef instead of memo

File: src/components/OAuthPopup.tsx [hunk 3]
```diff
@@ -21,9 +22,10 @@ export const OAuthPopup = ({
...
+
 	useEffect(() => {
-		if (didInit) return;
+		if (didInit) return () => {};
```
Comment (Cypherball) [inline src/components/OAuthPopup.tsx:37]: why this change?

File: src/components/OAuthPopup.tsx [hunk 4]
```diff
@@ -47,16 +48,30 @@ export const OAuthPopup = ({
 								? 'OAuth error: An error has occured.'
 								: 'OAuth error: State mismatch.'
 						}`;
+				channelPostMessage(channel, { type: OAUTH_RESPONSE, error: errorMessage });
 
-				openerPostMessage(opener, {
-					type: OAUTH_RESPONSE,
-					error: errorMessage,
-				});
+				console.error('message sent', errorMessage);
```
Comment (Cypherball) [inline src/components/OAuthPopup.tsx]: remove console logs

File: src/components/use-oauth2.ts [hunk 5]
```diff
@@ -45,8 +50,9 @@ export const useOAuth2 = <TData = TAuthTokenPayload>(props: TOauth2Props<TData>)
 		}
 	);
 
+	const channel = useMemo(() => new BroadcastChannel('refrens_oauth_channel'), []);
```
Comment (Cypherball) [inline src/components/use-oauth2.ts]: useRef

File: src/components/use-oauth2.ts [hunk 6]
```diff
@@ -137,30 +143,30 @@ export const useOAuth2 = <TData = TAuthTokenPayload>(props: TOauth2Props<TData>)
...
-				// Popup was closed before completing auth...
-				setUI((ui) => ({
-					...ui,
-					loading: false,
-				}));
-				console.warn('Warning: Popup was closed before completing authentication.');
-				cleanup(intervalRef, popupRef, handleBroadcastChannelMessage);
-			}
-		}, 250);
+		// intervalRef.current = setInterval(() => {
```
Comment (Cypherball) [inline src/components/use-oauth2.ts]: remove commented code

**Review by Cypherball — CHANGES_REQUESTED** [review]

File: src/components/OAuthPopup.tsx (hunk 3 above)
Comment (nidhi-bhanushali) [reply src/components/OAuthPopup.tsx:37]: below there is cleanup code in return block so for types this is required

**Review by Cypherball — APPROVED** [review]
